packages/eas-cli/src/project/remoteVersionSource.ts

```diff
@@ -45,10 +60,33 @@ export async function ensureVersionSourceIsRemoteAsync(
   Log.withTick('Updated eas.json');
 }
 
-export function validateBuildProfileVersionSettings(
+export async function validateBuildProfileVersionSettingsAsync(
   profileInfo: ProfileData<'build'>,
-  cliConfig: EasJson['cli']
-): void {
+  cliConfig: EasJson['cli'],
+  projectDir: string,
+  flags: BuildFlags
+): Promise<void> {
+  if (
+    cliConfig?.appVersionSource === undefined &&
+    profileInfo.profile.autoIncrement !== 'version'
+  ) {
+    if (profileInfo.profile.autoIncrement !== true) {
+      Log.log('The field "cli.appVersionSource" is not set, but it will be required in the future');
+      Log.log(
+        learnMore('https://docs.expo.dev/build-reference/app-versions/', {
+          learnMoreMessage: 'See the docs to learn more.',
+          dim: false,
+        })
+      );
```

Comment:
```suggestion
      Log.warn(`The field "cli.appVersionSource" is not set, but it will be required in the future. ${learnMore('https://docs.expo.dev/build-reference/app-versions/')}`);
```